README.md

```diff
@@ -22,11 +20,27 @@ Or install it yourself as:
 
 ## Usage
 
-TODO: Write usage instructions here
+`kubernetes-deploy <app's namespace> <kube context> --template-dir=DIR`
+
+Requirements:
+
+ - kubectl 1.5.1+ binary must be available in your path
+ - `ENV['KUBECONFIG']` must point to a valid kubeconfig file that includes the context you want to deploy to
+ - The target namespace must already exist in the target context
+ - `ENV['GOOGLE_APPLICATION_CREDENTIALS']` must point to the credentials for an authenticated service account if your user's auth provider is gcp
+ - `ENV['ENVIRONMENT']` must be set to use the default template path (`config/deploy/$ENVIRONMENT`) in the absence of the `--template-dir=DIR` option
+
 
 ## Development
 
-After checking out the repo, run `bin/setup` to install dependencies. Then, run `rake test` to run the tests. You can also run `bin/console` for an interactive prompt that will allow you to experiment.
+After checking out the repo, run `bin/setup` to install dependencies. You currently need to [manually install kubectl version 1.5.1 or higher](https://kubernetes.io/docs/user-guide/prereqs/) as well if you don't already have it.
+
+To run the tests:
+
+* [Install minikube](https://kubernetes.io/docs/getting-started-guides/minikube/#installation)
+* Start minikube (`minikube start [options]`)
+* Make sure you have a context named "minikube" in your kubeconfig. Minikube adds this context for you when you run `minikube start`; please do not rename it. You can check for it using `kubectl config get-contexts`.
+* Run `bundle exec rake test`
```

Comment:
Yes, but this is also an open source project and we should consider it may be used by someone who doesn't have `dev`.